vshard/storage/init.lua

```diff
@@ -292,8 +293,62 @@ local function bucket_generation_increment()
     M.bucket_generation_cond:broadcast()
 end
 
+local function bucket_commit_update(bucket)
+    local status = bucket.status
+    local bid = bucket.id
+    if status == consts.BUCKET.SENT or status == consts.BUCKET.GARBAGE then
+        local ref = M.bucket_refs[bid]
+        if ref ~= nil then
+            ref.ro_lock = true
+        end
+        return
+    end
+    if status == consts.BUCKET.RECEIVING then
+        -- It shouldn't have a ref, but prepare to anything. Could be, for
+        -- example, ACTIVE changed into RECEIVING manually via
+        -- _bucket:replace().
+        if M.bucket_refs[bid] ~= nil then
+            M.bucket_refs[bid] = nil
+        end
+        return
+    end
+    if status == consts.BUCKET.PINNED or status == consts.BUCKET.ACTIVE then
+        local ref = M.bucket_refs[bid]
+        if ref ~= nil then
+            -- Lock shouldn't be ever set here. But could be that an admin
+            -- decided to restore a SENT bucket back into ACTIVE state due to
+            -- any reason.
+            ref.ro_lock = false
+        end
+        return
+    end
+    -- Ignore unknown statuses. Could be set no idea how, but just ignore.
```

Comment:
Maybe we can at least log them?

Reply:
At first I thought you are right and went to add the log, but then remembered that we don't have a ratelimited log in Lua (not publicly). That means if we have a massive bucket status rename, we will produce potentially gigabytes of logs all over the cluster. I propose to keep it like that, it is at least not worse that it was. And in future commits I plan to add validation into `on_replace`, so a bad status won't even be able to commit.